source/extensions/access_loggers/fluentd/fluentd_access_log_impl.cc

```diff
@@ -102,6 +111,24 @@ void FluentdAccessLoggerImpl::flush() {
   clearBuffer();
 }
 
+void FluentdAccessLoggerImpl::connect() {
+  connect_attempts_++;
+  if (!client_->connect()) {
+    ENVOY_LOG(debug, "no healthy upstream");
+    setDisconnected();
+    return;
```

Comment:
It's complete possible scenario where all endpoints are gone and then them go back after some time.

But this won't work in that case.

Reply:
I think If you support the `reconnect_interval`, the reconnect timer could resolve this issue naturally?